atom/browser/api/atom_api_web_contents.cc

```diff
@@ -537,6 +538,7 @@ void WebContents::CompleteInit(v8::Isolate* isolate,
 
   InitWithWebContents(web_contents, GetBrowserContext());
   managed_web_contents()->GetView()->SetDelegate(this);
+  SetSafeBrowsingObserver();
```

Comment:
since we only care about tabs, the observer should probably be added inside `IsGuest()` below

Reply:
`IsGuest()` is a const function, so initialization will fail:

```
../../electron/atom/browser/api/atom_api_web_contents.cc:2487:73: error: cannot initialize a parameter of type 'safe_browsing::SafeBrowsingUIManager::Observer *' with an rvalue of type 'const atom::api::WebContents *'
  g_browser_process->safe_browsing_service()->ui_manager()->AddObserver(this);
```

Moving the code to AddObserver to the `IsGuest()` block in `CompleteInit`